### PR TITLE
Overhaul the deployment

### PR DESCRIPTION
**Pull request by supertassu**
- pipeline: Remove folder
- build: Use Wikimedia-provided Go image
- build: Upgrade dependencies
- build: Fix Go import path to match reality
- server: Properly fail when certificates fail to load
- deployment: Deploy using Helm
- Update README for Helm deployment
- deployment: use cert-manager certificates

### CODE REVIEW

File: deployment/utils/local-values.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
```
Comment (NdibeRaymond) [inline deployment/utils/local-values.sh:1]: ./deploy.sh local errors out with `./deploy.sh: line 20: ./deployment/utils/local-values.sh: Permission denied`. We probably need to add execute permission.

File: deploy.sh [hunk 2]
```diff
@@ -1,147 +1,34 @@
...
+kubectl create namespace buildpack-admission --dry-run=client -o yaml | kubectl apply -f -
 
+if [ "$DEPLOY_ENVIRONMENT" = "local" ]; then
+	DEV_DOMAIN_IP="${DEV_DOMAIN_IP}" ./deployment/utils/local-values.sh
+fi
 
-check_environment() {
-    # verify that the proper environment is passed
-    local environment="${1?No environment passed}"
+./deployment/utils/regenerate-certs.sh
```
Comment (NdibeRaymond) [inline deploy.sh]: since we deleted the file regenerate-certs we should also remove this line

**Comment by NdibeRaymond** [comment]
Thanks for working on this @supertassu !

**Review by NdibeRaymond — CHANGES_REQUESTED** [review]

**Comment by supertassu** [comment]
Thanks @NdibeRaymond, fixed both of those.

**Review by NdibeRaymond — APPROVED** [review]

**Comment by NdibeRaymond** [comment]
Hello @supertassu , I noticed another thing. I couldn't figure out the root on time yesterday so didn't report it. maybe you can help to track it down. after the recent fixes, running `./deloy.sh local` ends with an error:

```
 ./deploy.sh local
Building dependency release=buildpack-admission, chart=chart
Comparing release=buildpack-admission, chart=chart
in deployment/helmfile.yaml: command "/usr/sbin/helm" exited with non-zero status:

PATH:
  /usr/sbin/helm

ARGS:
  0: helm (4 bytes)
  1: diff (4 bytes)
  2: upgrade (7 bytes)
  3: --reset-values (14 bytes)
  4: --allow-unreleased (18 bytes)
  5: buildpack-admission (19 bytes)
  6: chart (5 bytes)
  7: --namespace (11 bytes)
  8: buildpack-admission (19 bytes)
  9: --values (8 bytes)
  10: /tmp/values731471351 (20 bytes)
  11: --detailed-exitcode (19 bytes)

ERROR:
  exit status 1

EXIT STATUS
  1

STDERR:
  Error: Failed to render chart: exit status 1: Error: unable to build kubernetes objects from release manifest: [resource mapping not found for name: "buildpack-admission-certs" namespace: "" from "": no matches for kind "Certificate" in version "cert-manager.io/v1"
  ensure CRDs are installed first, resource mapping not found for name: "buildpack-admission-certs" namespace: "" from "": no matches for kind "Issuer" in version "cert-manager.io/v1"
  ensure CRDs are installed first]
  Error: plugin "diff" exited with error

COMBINED OUTPUT:
  ********************
  	Release was not present in Helm.  Diff will show entire contents as new.
  ********************
  Error: Failed to render chart: exit status 1: Error: unable to build kubernetes objects from release manifest: [resource mapping not found for name: "buildpack-admission-certs" namespace: "" from "": no matches for kind "Certificate" in version "cert-manager.io/v1"
  ensure CRDs are installed first, resource mapping not found for name: "buildpack-admission-certs" namespace: "" from "": no matches for kind "Issuer" in version "cert-manager.io/v1"
  ensure CRDs are installed first]
  Error: plugin "diff" exited with error

```

**Comment by supertassu** [comment]
The key is this line:
```
Error: Failed to render chart: exit status 1: Error: unable to build kubernetes objects from release manifest: [resource mapping not found for name: "buildpack-admission-certs" namespace: "" from "": no matches for kind "Certificate" in version "cert-manager.io/v1"
```
That suggests you don't have [cert-manager](https://cert-manager.io/) installed. [lima-kilo](https://gitlab.wikimedia.org/repos/cloud/toolforge/lima-kilo/) will install it automatically for you, or you can do it manually from [our deployment repository](https://gitlab.wikimedia.org/repos/cloud/toolforge/cert-manager).

**Comment by NdibeRaymond** [comment]
ok thanks @supertassu , I should get that sorted then

**Comment by NdibeRaymond** [comment]
update: I think I'd have to install the cert-manager manually. limo-kilo doesn't do that automatically
or maybe I have the wrong lima-kilo repo

**Pull request merged**